django/contrib/contenttypes/models.py

```diff
@@ -31,6 +31,12 @@ def _get_from_cache(self, opts):
         key = (opts.app_label, opts.model_name)
         return self._cache[self.db][key]
 
+    def get_for_permission_by_model(self, model):
```

Comment:
This name doesn't read right.

Reply:
I agree, I initially wanted to use `Permission.objects.get_content_type_for_model` instead, but I ran into issues because the imported `ContentType` in `django.contrib.auth.models` is different than the one in [`django/contrib/auth/management/__init__.py`](https://github.com/django/django/pull/10381/files#diff-23ce0f3ba59117c95d173175f52d60c7R47):

```python
app_config = apps.get_app_config(app_label)
ContentType = apps.get_model('contenttypes', 'ContentType')
```

If someone can help me figure this out then I think it would be a much better solution to **a.** Have a better name, and **b.** Living on the `Permission` model.

I'm open to other suggestions too of course.

Reply:
@auvipy It makes me think that you are getting a `permission_type` which is unclear what that could be... If there is no good name for it, I can also get rid of it and call `get_for_model(Model, for_concrete_model=False)` explicitly. It's not something that should be used a lot and I can add an entry in the docs for proxy models, what do you think?

Reply:
> I can also get rid of it and call get_for_model(Model, for_concrete_model=False) explicitly.

Yes. I think this. 

Outside of the tests, there's just a single call-site. I don't see that merits a _convenience wrapper_, rather than just passing the `kwarg`.